src/syrupy/extensions/single_file.py

```diff
@@ -82,9 +82,10 @@ def _read_snapshot_collection(
     ) -> "SnapshotCollection":
         file_ext_len = len(self._file_extension) + 1 if self._file_extension else 0
         filename_wo_ext = snapshot_location[:-file_ext_len]
+        basename = Path(filename_wo_ext).parts[-1]
 
         snapshot_collection = SnapshotCollection(location=snapshot_location)
-        snapshot_collection.add(Snapshot(name=Path(filename_wo_ext).stem))
```

Comment:
We're already stripping the suffix on L84. We should not be doing this twice, otherwise our class method naming convention `TestClass.test_method.json` will turn into `TestClass`.